pwnlib/commandline/libcdb.py

```diff
@@ -136,18 +152,67 @@
 
 common_symbols = ['dup2', 'printf', 'puts', 'read', 'system', 'write']
 
-def find_libc(params):
+
+def find_in_online_mode(params):
     import requests
-    url    = "https://libc.rip/api/find"
+    url = "https://libc.rip/api/find"
     result = requests.post(url, json=params, timeout=20)
     log.debug('Request: %s', params)
     log.debug('Result: %s', result.json())
     if result.status_code != 200 or len(result.json()) == 0:
         log.failure("Could not find libc for %s on libc.rip", params)
         return []
-
     return result.json()
 
+
+def find_in_offline_mode(params):
+    # lookup parser
+    if params.get("symbols"):
+        matching_libcs = libcdb.find_local_libc(params)
+        return matching_libcs if matching_libcs else []
+
+    # hash parser
+    hash_type, hash_value = list(params.items())[0]
+
+    local_db = libcdb._fetch_local_database_path()
```

Comment:
Can you access the other new libcdb functions directly that way as well without exporting them? `get_libc_info` etc aren't useful outside the libcdb module.